parastell/nwl_utils.py

```diff
@@ -215,192 +218,217 @@ def extract_coords(source_file):
-            at once
-        num_threads (int): number of threads to use for NWL calculations,
-            defaults to 1.
+        toroidal_extent (float): toroidal extent of model [deg].
+        neutron_power (float): reference neutron power [MW].
+        num_toroidal_bins (int): number of toroidal angle bins (defaults to 101).
+        num_poloidal_bins (int): number of poloidal angle bins (defaults to 101).
+        conv_tol (float): tolerence for convergence in poloidal angle root-
+            finding routine. Must lie in range (0.0, 1.0]. Smaller values
+            correspond to a stricter tolerance. This parameter corresponds to
+            the hyperrectangle volume tolerance defining the termination
+            criterion for SciPy's DIRECT algorithm. Once the volume of the
+            hyperrectangle containing the lowest function value (root-finding
+            residual) falls below this tolerance, root-finding will terminate.
+        num_batches (int): number of batches across which crossing coordinates
+            will be solved (defaults to 1). Helps alleviate memory burden.
+        num_crossings (int): number of crossings to use from the surface source
+            (defaults to None). If None, all crossings will be used.
+        num_threads (int): number of threads to use for parallelizing
+            coordinate-solving routine (defaults to None). If None, the maximum
+            number of threads will be used.
 
     Returns:
         nwl_mat (numpy array): array used to create the NWL plot
-        phi_pts (numpy array): phi axis of NWL plot
-        theta_pts (numpy array): theta axis of NWL plot
-        area_array (numpy array): area array used to normalize nwl_mat
+        toroidal_centroids (numpy array): phi axis of NWL plot
+        poloidal_centroids (numpy array): theta axis of NWL plot
+        area_mat (numpy array): area array used to normalize nwl_mat
     """
-    tor_ext = np.deg2rad(tor_ext)
-    pol_ext = np.deg2rad(pol_ext)
+    logger = log.check_init(logger)
+
+    toroidal_extent = np.deg2rad(toroidal_extent)
+    poloidal_extent = 2 * np.pi
 
-    coords = extract_coords(source_file)
+    if not num_threads:
+        num_threads = os.cpu_count()
 
+    coords = extract_surface_crossings(source_file)
     if num_crossings is not None:
         coords = coords[0:num_crossings]
 
-    vmec = read_vmec.VMECData(plas_eq)
+    num_particles = len(coords)
 
-    phi_coords = []
-    theta_coords = []
+    toroidal_angles = []
+    poloidal_angles = []
 
-    if chunk_size is None:
-        chunk_size = len(coords)
+    batch_size = math.ceil(num_particles / num_batches)
 
-    chunks = math.ceil(len(coords) / chunk_size)
+    iterator = 1
 
-    for i in range(chunks):
-        phi_coord_subset, theta_coord_subset = flux_coords(
-            plas_eq,
+    for batch_start in range(0, num_particles, batch_size):
+        iterator += 1
```

Comment:
You don't need `iterator` if you use

```suggestion
        batch_num = batch_start/batch_size + 1
```

Reply:
I was thinking of doing something like that but I figured if we're going to save the value anyways, something like `iterator += 1` is more readable? Although in either case, `batch_num` is probably the better variable name to use. What do you think?

Reply:
If you want to use a variable that iterates like this, then you can/should use `enumerate`